packages/blade/src/components/ChatMessage/_decision.md

```diff
@@ -0,0 +1,85 @@
+| senderType          | 'self' or 'other'                              | self   | No       | we will add  different styles based on this bubble                              |
+| isLoading              | Boolean                                     | false   | No       | If the message is loading, we will add a loading animation to the chat bubble                                                                                   |
+| validationState                | 'error' or 'none'                              | none   | No       |  if validation state is error we will show error decoration and message|
+| errorText              | String                                      | null    | No       | If the message is an error message, we will show the error message in the chat bubble                                                                                   |
+| onClick           | Function                                    | null    | No       | callback to be called when ever component is clicked.                                                                                                                          |
+| footerActions        | ReactNode | null    | No       | if this is passed we will render this at the end of chat bubble                               |
+| children               |  'ReactNode' or 'string'                          | null    | yes      | The children that will be rendered inside the chat bubble. can be react node or a string                                                                        |
+| leading             | ReactNode                                   | null    | No       | will be displayed only in case  if message is other and also, will contain animation by default                                   |
+|loadingText | String | null | No | if loading is true, we will show this text in the chat bubble |
+
+```tsx
+type ChatMessageProps = {
+  messageType?: 'last' | 'default';
+  senderType?: 'self' | 'other';
+  isLoading?: boolean;
+  validationState?: 'error' | 'none';
+  errorText?: string;
+  onClick?: () => void;
+  footerActions?: ReactNode;
+  children: ReactNode | string;
+  leading?: ReactNode;
+  loadingText?: string;
+};
+```
+
+## API
+
+```tsx
+
+// for animation
+<Move>
+  <ChatMessage senderType="self">Demo Text</ChatMessage>
+</Move>
+
+// with error
+<ChatMessage validationState="error" errorText="Error Message">Demo Text</ChatMessage>
+
+// with card
+<ChatMessage><Card></Card></ChatMessage>
+
+
+<ChatMessage><Markdown>  Demo Text </Markdown></ChatMessage>
+<ChatMessage><Markdown> Demo Text </Markdown></ChatMessage>
+```
+
+## Alternative API
+
+```tsx
+// for animation
+<Move>
+<ChatMessage message="Demo Text"/>
+</Move>
+
+// with card
+<ChatMessage cardBody={<SomeComponent/>} />
+
+
+
+// Markdown
+<ChatMessage markdown="# this is markdown" />
+```
+
+## Open Questions
+
```

Comment:
So normally we keep the questions in open questions and add conclusion and discussion points based on discussion so that we have reasoning in place if we want to find it later